plugins/outputs/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,246 @@
+package elasticsearch
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"gopkg.in/olivere/elastic.v5"
+)
+
+type Elasticsearch struct {
+	URLs                []string `toml:"urls"`
+	IndexName           string
+	Username            string
+	Password            string
+	EnableSniffer       bool
+	HealthCheckInterval int
+	ManageTemplate      bool
+	TemplateName        string
+	OverwriteTemplate   bool
+	Client              *elastic.Client
+}
+
+var sampleConfig = `
+  ## The full HTTP endpoint URL for your Elasticsearch instance
+  ## Multiple urls can be specified as part of the same cluster,
+  ## this means that only ONE of the urls will be written to each interval.
+  urls = [ "http://node1.es.example.com:9200" ] # required.
+  ## Set to true to ask Elasticsearch a list of all cluster nodes,
+  ## thus it is not necessary to list all nodes in the urls config option
+  enable_sniffer = true
+  ## Set the interval to check if the nodes are available, in seconds
+  ## Setting to 0 will disable the health check (not recommended in production)
+  health_check_interval = 10
+  ## HTTP basic authentication details (eg. when using Shield)
+  # username = "telegraf"
+  # password = "mypassword"
+
+  # Index Config
+  ## The target index for metrics (telegraf will create it if not exists)
+  ## You can use a different index per time frame using the patterns below
+  # %Y - year (2016)
+  # %y - last two digits of year (00..99)
+  # %m - month (01..12)
+  # %d - day of month (e.g., 01)
+  # %H - hour (00..23)
+  index_name = "telegraf-%Y.%m.%d" # required.
+
+  ## Template Config
```

Comment:
can you provide more information on the template management? doesn't have to be here, maybe in the README would be sufficient.

links to elasticsearch documentation would be extra good.